src/client.ts

```diff
@@ -3764,6 +3764,23 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         }
     }
 
+    public async deleteAccountData(eventType: string): Promise<void> {
+        const supportsMsc3391DeleteAccountData =
+            this.canSupport.get(Feature.AccountDataDeletion) !== ServerSupport.Unsupported;
+        // if deletion is not supported overwrite with empty content
+        if (!supportsMsc3391DeleteAccountData) {
+            await this.setAccountData(eventType, {});
+            return;
+        }
+        const path = utils.encodeUri("/user/$userId/account_data/$type", {
+            $userId: this.credentials.userId!,
```

Comment:
This is a public method and can be called when no user is logged in. Means the `!` is probably incorrect here. We should handle the case when it is null.

Reply:
True, I just copied this part from the `setAccountData` method, which I assume relies on `authedRequest` to fail nicely when there is no logged in user. Added `getSafeUserId` to use here